File: cla-backend-go/events/eventtypes.go
// Copyright The Linux Foundation and each contributor to CommunityBridge.
// SPDX-License-Identifier: MIT

package events

// event types
const (
	// CreateUser event type
	CreateUser = "Create User"
	// UpdateUser event type
	UpdateUser = "Update User"
	// DeleteUser event type
	DeleteUser = "Delete User"

	// CreateTemplate event type
	CreateTemplate = "Create Template"

	// AddGithubOrgToWL event type
	AddGithubOrgToWL = "Add GH Org To WL"
	// DeleteGithubOrgFromWL event type
	DeleteGithubOrgFromWL = "Delete GH Org From WL"

	// CreateCCLAWhitelistRequest event type
	CreateCCLAWhitelistRequest = "Create CCLA WL Request"
	// DeleteCCLAWhitelistRequest event type
	DeleteCCLAWhitelistRequest = "Delete CCLA WL Request"

	// AddUserToCompanyACL event type
	AddUserToCompanyACL = "Add User To Company ACL"
	// DeleteUserFromCompanyACL event type
	//DeleteUserFromCompanyACL = "Delete User From Company ACL"

	// DeletePendingInvite event type
	DeletePendingInvite    = "Delete Pending Invite"
	AddGithubRepository    = "Add Github Repository"
	DeleteGithubRepository = "Delete Github Repository"
)

File: cla-backend/cla/models/event_types.py
# Copyright The Linux Foundation and each contributor to CommunityBridge.
# SPDX-License-Identifier: MIT

from enum import Enum


class EventType(Enum):
    """
    Enumeraters representing type of CLA events
    across projects, users, signatures, whitelists

    """
    CreateUser = "Create User"
    UpdateUser = "Update User"
    DeleteUser = "Delete User"
    CreateProject = "Create Project"
    UpdateProject = "Update Project"
    DeleteProject = "Delete Project"
    MigrateProjectSFID = "Migrate Project SFID"
    CreateCompany = "Create Company"
    DeleteCompany = "Delete Company"
    UpdateCompany = "Update Company"
    CreateProjectDocument = "Create Project Document"
    CreateProjectDocumentTemplate = "Create Project Document with Template"
    DeleteProjectDocument = "Delete Project Document"
    AddPermission = "Add Permission"
    RemovePermission = "Remove Pemrission"
    AddProjectManager = "Add Project Manager"
    RemoveProjectManager = "Remove Project Manager"
    RequestCompanyWL = "Request Company Whitelist"
    InviteAdmin = "Invite Admin"
    RequestCCLA = "Request Company CCLA"
    RequestCompanyAdmin = "Request Company Admin access"
    AddCompanyPermission = "Add Company Permissions"
    RemoveCompanyPermission = "Remove Company Permissions"
    CreateSignature = "Create Signature"
    DeleteSignature = "Delete Signature"
    UpdateSignature = "Update Signature"
    AddCLAManager = "Add CLA Manager"
    RemoveCLAManager = "Remove CLA Manager"
    NotifyWLChange = "Notify WL Change"
    UserAssociatedWithCompany = "User associated with company"
    EmployeeSignatureCreated = "Employee signature created"
    EmployeeSignatureDisapproved = "Employee signature disapproved"
    IndividualSignatureSigned = "Individual signature signed"
    EmployeeSignatureSigned = "Employee signature signed"
    CompanySignatureSigned = "Company signature signed"
    RepositoryAdded = "Repository Added"
    RepositoryRemoved = "Repository Removed"
    RepositoryDisable = "Repository Disabled"
    RepositoryEnabled = "Repository Enabled"
